api/v1/flow/flow.proto

```diff
@@ -311,6 +311,10 @@ message TCP {
     uint32 source_port = 1;
     uint32 destination_port = 2;
     TCPFlags flags = 3;
+    // source_port_xlated is the post-translation source port when the flow was
+    // SNATed. When "source_port_xlated" is set, the "source_port" field is
+    // populated with the pre-translation source port.
+    uint32 source_port_xlated = 4;
```

Comment:
Will defer to @rolinh on this one, but would it make sense to put this in the l4 message to avoid duplicating it?